rakefile.netstd.rb

```diff
@@ -0,0 +1,400 @@
+
+  client = Octokit::Client.new(:netrc => true)
+
+  puts "Creating branch '#{new_branch}'..."
+  `git checkout -b #{new_branch}`
+  puts "Created branch '#{new_branch}'."
+
+  puts "Setting version to '#{new_version}' in '#{assembly_info}'..."
+  Rake::Task["set_version_in_assemblyinfo"].invoke(new_version)
+  puts "Set version to '#{new_version}' in '#{assembly_info}'."
+
+  puts "Committing '#{assembly_info}'..."
+  `git commit -m "setting version to #{new_version}" #{assembly_info}`
+  puts "Committed '#{assembly_info}'."
+
+  puts "Pushing '#{new_branch}' to origin..."
+  `git push origin #{new_branch}"`
+  puts "Pushed '#{new_branch}' to origin."
+
+  puts "Creating pull request..."
+  pull_request = client.create_pull_request(
+    repo,
+    "master",
+    "#{client::user.login}:#{new_branch}",
+    "set version to #{new_version} for next release",
+    "preparing for #{new_version}"
+  )
+  puts "Created pull request \##{pull_request.number} '#{pull_request.title}'."
+
+  puts "Creating milestone '#{new_version}'..."
+  milestone = client.create_milestone(
+    repo,
+    new_version,
+    :description => new_version + ' release'
+    )
+  puts "Created milestone '#{new_version}'."
+
+  create_release(
+                 client,
+                 repo,
+                 milestone,
+                 new_version,
+                 release_body,
+                 release_issue_common_steps + next_version_steps,
+                 release_issue_labels)
+end
+
+desc "Create GitHub Release and release checklist issue for pre-release build"
+task :pre_release, :version_suffix do |asm, args|
+  version_suffix = args.version_suffix or
+    fail 'ERROR: A version suffix is required, e.g.: rake pre_release[beta001]'
+
+  require 'octokit'
+
+  ssl_cert_file = get_temp_ssl_cert_file(ssl_cert_file_url)
+
+  client = Octokit::Client.new(:netrc => true)
+
+  milestone = client
+    .list_milestones(repo, :state => 'open')
+    .select { |m| m.title == version }
+    .first or
+    raise "ERROR: can't find existing milestone #{version}"
+
+  create_release(
+                 client,
+                 repo,
+                 milestone,
+                 "#{version}-#{version_suffix}",
+                 release_body,
+                 release_issue_common_steps,
+                 release_issue_labels)
+end
+
+desc "Update assembly info"
+assemblyinfo :set_version_in_assemblyinfo, :new_version do |asm, args|
+  new_version = args.new_version
+
+  # not using asm.version and asm.file_version due to StyleCop violations
+  asm.custom_attributes = {
+    :AssemblyVersion => new_version,
+    :AssemblyFileVersion => new_version,
+    :AssemblyInformationalVersion => new_version
+  }
+  asm.input_file = assembly_info
+  asm.output_file = assembly_info
+end
+
+desc "Build solution"
+task :build => [:clean, :restore, logs] do
+  netstd_unit_test_directories.each do | test_directory |
+    restore_dependencies_for_project test_directory
+  end

```

Comment:
`dotnet restore` is not sufficient.

Initial attempt to move this block to the `restore` task failed miserably.
But it was because I had a malformed (although still technically valid?) task definition. That's probably what happened to me the first time. Turns out I can move it. I'm glad you said something.